scripts/firedrake-zenodo

```diff
@@ -23,6 +24,7 @@ descriptions = OrderedDict([
 projects = dict(
     [("firedrake", "firedrakeproject"),
      ("PyOP2", "OP2"),
+     ("SLOPE", "coneoproject"),
      ("tsfc", "firedrakeproject"),
      ("COFFEE", "coneoproject"),
      ("ufl", "firedrakeproject"),
```

Comment:
This barfs if the user calls `firedrake-zenodo` but has not installed SLOPE, which is an optional module.  Maybe something like:

```patch
diff --git a/scripts/firedrake-zenodo b/scripts/firedrake-zenodo
index 1c6c480..24a7ae7 100755
--- a/scripts/firedrake-zenodo
+++ b/scripts/firedrake-zenodo
@@ -34,6 +34,8 @@ projects = dict(
 
 components = descriptions.keys()
 
+optional_components = ("SLOPE", )
+
 parser = ArgumentParser(description="""Create Zenodo DOIs for specific versions of Firedrake components.
 
 If you are a Firedrake user, this script creates a JSON file encoding
@@ -135,8 +137,12 @@ def collect_repo_shas():
                 sys.exit(0)
             shas[component] = check_output(["git", "rev-parse", "HEAD"]).strip()
         except:
-            log.error("Failed to retrieve git hash for %s" % component)
-            raise
+            if component in optional_components:
+                log.warning("Failed to retrieve git hash for optional "
+                            "component '%s', continuing without it" % component)
+            else:
+                log.error("Failed to retrieve git hash for %s" % component)
+                raise
     return shas
 
 if args.release_tag:
```